File: src/libs/parsers/RectangleParser.ts
import { Figure } from "../figures/Figure";
import { Rectangle } from "../figures/Rectangle";
import { Point } from "../figures/Point";

class RectangleParser {

    public getData(json:any):Rectangle {
        let rectangle:Rectangle;
        try {
            rectangle = new Rectangle(new Point(json.startX, json.startY), json.height, json.width, json.fillColor);
        } catch (err) {
            rectangle = null;
        }
        return rectangle;
    }


}

export { RectangleParser };

File: src/libs/parsers/RectangleParser.java
package org.ambrozy.figurepainter.parsers;

import org.ambrozy.figurepainter.figures.Figure;
import org.ambrozy.figurepainter.figures.Line;
import org.ambrozy.figurepainter.figures.Point;
import org.ambrozy.figurepainter.figures.Rectangle;

import java.util.Scanner;

public class RectangleParser {
    private final Scanner scanner;

    public RectangleParser(Scanner s2) {
        this.scanner = s2;
    }

    public Figure getData() {
        Rectangle rectangle;
        try {
            Point start = new Point(Double.parseDouble(scanner.next()), Double.parseDouble(scanner.next()));
            Double width = Double.parseDouble(scanner.next());
            Double height = Double.parseDouble(scanner.next());
            rectangle = new Rectangle(start, width, height);
        } catch(Exception e) {
            return null;
        }

        return rectangle;
    }
}
